Tests/MLXTests/OpsTests.swift

```diff
@@ -46,4 +46,36 @@ class OpsTests: XCTestCase {
         assertEqual(b, expected)
     }
 
+    func testConvertScalarInt() {
+        let a = MLXArray(0 ..< 10)
+        let b = a .< (a + 1)
+        let c = b * 25
+        XCTAssertEqual(b.dtype, .bool)
+        XCTAssertEqual(c.dtype, .int32)
```

Comment:
I discovered that types were not being promoted from bool -> int/float correctly.